src/gc/gcrecord.h

```diff
@@ -19,7 +19,7 @@ Module Name:
 // We pack the dynamic tuning for deciding which gen to condemn in a DWORD.
 // We assume that 2 bits are enough to represent the generation. 
 #define bits_generation 2
-#define generation_mask (~(~0 << bits_generation))
+#define generation_mask (~(~((unsigned int)0) << bits_generation))
```

Comment:
A fix for this (using "u" suffix instead of the cast) was already merged in as #1650 a day ago